oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/ConfigHelper.java

```diff
@@ -48,4 +52,11 @@ public static boolean getSystemPropertyAsBoolean(String name, boolean defaultVal
         LOG.info("Config {}={}", name, value);
         return value;
     }
+
+    public static List<String> getSystemPropertyAsStringList(String name, String defaultValue, String separator) {
+        String result = System.getProperty(name, defaultValue);
+        List<String> parts = result.isBlank() ? List.of() : Arrays.stream(result.split(separator)).map(String::trim).collect(Collectors.toList());
```

Comment:
What about to methods: `getSystemProperty` and `splitIgnoreSpace`? I see the code is quite lenient in what it accepts (isBlank vs isEmpty, trim). For user-facing configuration, I think it's the right thing to do; we ran into problems because people added spaces eg. "type" = "lucene " (trailing space). 

(I think it is not necessary in this case because this is not end-user facing but only for internal use. For internal use, I think non-lenient behavior is better because it results in shorter, easier to read code. But it's not that bad here, so I think it's fine.)

But more importantly, what if spaces have meaning? The method name `getSystemPropertyAsStringList` doesn't convey what it does, and there is no Javadoc.

What about renaming "separator" -> "separatorRegex", or then quote the separator (using Pattern.quote I think)?

Reply:
To reduce the chance of errors, I changed separator to be a `char` and use a method from Commons Lang that splits using char, instead of regex. I created a separate method as private. Maybe in the future could be made made into an utility method, in case it does not already exist.